.github/workflows/ci.yml

```diff
@@ -30,6 +29,8 @@ jobs:
       uses: actions/setup-python@v2
       with:
         python-version: "3.9"
+    - name: Lint with Pre-commit
+      uses: pre-commit/action@v2.0.3
```

Comment:
Does this modify the code, or just run flake/black/etc checks and report an error if they haven't been run? If it *does* modify - who is the author of the commit?

Reply:
Exits with an error if pre-commit would need to modify a file.

Reply:
Done, moved pre-commit to the first action run, then beefore follows. Another option is, we could make use of https://pre-commit.ci. An advantage is it runs faster and automatically creates PRs with the pre-commit hooks are out of date. This could be a future thing we look at, though.